pkg/reconciler/broker/broker.go

```diff
@@ -320,12 +321,15 @@ func (r *Reconciler) deleteDecouplingTopicAndSubscription(ctx context.Context, b
 		return err
 	}
 
-	client, err := r.CreateClientFn(ctx, projectID)
-	if err != nil {
-		logger.Error("Failed to create Pub/Sub client", zap.Error(err))
-		return err
+	client := r.pubsubClient
+	if client == nil {
+		client, err := pubsub.NewClient(ctx, projectID)
+		if err != nil {
+			logger.Error("Failed to create Pub/Sub client", zap.Error(err))
+			return err
+		}
+		defer client.Close()
```

Comment:
Curious why creating client here rather than in the controller.go and reuse the same client over and over? Similar question for several other places.

Reply:
If project id ever changes between reconciles, then a single client wouldn't work, but we don't support that today. Can we properly close the client if it's created by the controller?

Reply:
It seems it's recommended to reuse pubsub client: https://pkg.go.dev/cloud.google.com/go/pubsub?tab=doc#Client
Maybe better to only recreate the client if the project is changed. But I don't know how to handle client close properly if we reuse the pubsub client.

Reply:
One thing I can think of is to override https://github.com/google/knative-gcp/blob/12200f7088cd728b7e1a683fe5b6db80b7a36468/vendor/knative.dev/pkg/controller/controller.go#L354

Reply:
Just pushed a change to use a single pubsub client for all reconciles if possible. If not possible the reconcilers fall back to the original behavior. PTAL.